src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,143 @@
+import { useEffect, useState } from 'react';
+import { useLocation, useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { getSearchWith } from '../utils/searchHelper';
+import { SortType } from '../types/SortType';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[] | null>(null);
+  const [copyPeople, setCopyPeople] = useState<Person[] | null>(null);
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+  const location = useLocation();
+
+  const onQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchParams(
+      getSearchWith(searchParams, { query: event.target.value || null }),
+    );
+  };
+
+  const loadPeople = async () => {
+    setIsLoading(true);
+    try {
+      const loadedPeople = await getPeople();
+
+      setPeople(loadedPeople);
+      setCopyPeople(loadedPeople);
+    } catch {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadPeople();
+  }, []);
+
+  useEffect(() => {
+    const visiblesPeople = copyPeople?.filter(person => {
+      switch (true) {
+        case sex.length > 0:
+          return person.sex === sex;
+
+        case query.length > 0:
+          return person.name.includes(query)
+            || person.fatherName?.includes(query)
+            || person.motherName?.includes(query);
+
+        case centuries.length > 0:
+          return centuries.includes((Math.ceil(person.born / 100)).toString());
+
+        default:
+          return person;
+      }
+    });
```

Comment:
Move this value from the useEffect. You can use useMemo to memorize this value if you need to